ipn/ipnstate/ipnstate.go

```diff
@@ -270,6 +270,12 @@ type PeerStatus struct {
 	// PeerAPIURL are the URLs of the node's PeerAPI servers.
 	PeerAPIURL []string
 
+	// CanReceiveFiles indicates this peer is a valid file-transfer target.
+	CanReceiveFiles bool
+
+	// Reason why this peer cannot receive files. Empty if CanReceiveFiles=true
+	NoFileSharingReason string
```

Comment:
Consider collapsing this and the above `bool` into a single `TaildropTargetStatus`.

Also, instead of making this a "string", consider making it an `int` enum, like this:

```
type TaildropTargetStatus int

const (
    TaildropTargetAvailable = iota
    TaildropTargetOffline
    TaildropTargetUnsupportedOS
    TaildropTargetNoPeerAPI
    TaildropTargetOwnedByOtherUser
)
```